core/src/main/java/edu/wpi/grip/core/util/Holder.java

```diff
@@ -0,0 +1,42 @@
+
+package edu.wpi.grip.core.util;
+
+/**
+ * Simple thread-safe class to allow for setting values from lambdas.
+ */
+public final class Holder<T> {
+
+    private T value;
+
+    /**
+     * Creates a new holder with the given default value.
+     */
+    public Holder(T value) {
+        if (value == null) {
+            throw new NullPointerException("Held value cannot be null");
+        }
+        this.value = value;
+    }
+
+    /**
+     * Gets the value of this holder. This is a thread-safe operation.
+     *
+     * @return the value of this holder.
+     */
+    public synchronized T get() {
+        return value;
+    }
+
+    /**
+     * Sets the value of this holder. This is a thread-safe operation.
+     *
+     * @param value the new value of this holder.
+     */
+    public synchronized void set(T value) {
+        if (value == null) {
+            throw new NullPointerException("Held value cannot be null");
+        }
```

Comment:
This can be replaced with the google preconditions check.
https://github.com/google/guava/wiki/PreconditionsExplained

This whole method could become:
```java
public synchronized void set(T value) {
    this.value = checkNotNull(value, "Held value cannot be null");
}
```